src/components/SettingsModal.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useContext, useEffect } from "react";
+import reactDom from "react-dom";
+import classnames from "classnames";
+import { v4 as uuidv4 } from "uuid";
+import { ModalContext } from "./ClockFace.js"
+import "./SettingsModal.scss";
+
+const SettingsModal = () => {
+  const { isModalOpen, setIsModalOpen, sessions, setSessions } = useContext(ModalContext);
+  const [localSession, setLocalSession] = useState(sessions);
+  const [hasSettingsChanged, setHasSettingsChanged] = useState(false);
+
+  useEffect(() => {
+    if (isModalOpen) {
+      setLocalSession(sessions);
```

Comment:
This feels kinda weird, because the default of the localSession hook is already sessions. But I have to set it again here to be sessions when user clicks the Close button without saving modal settings. Because when opening the modal again, localSession values need to be updated back to the real sessions setting.
Or else, the modal's localSession values will stay at the unsaved values.